include/deal.II/lac/trilinos_tpetra_block_sparse_matrix.templates.h

```diff
@@ -0,0 +1,469 @@
+
+      const size_type n_block_rows = tpetra_maps.size();
+      (void)n_block_rows;
+
+      Assert(n_block_rows == block_sparsity_pattern.n_block_rows(),
+             ExcDimensionMismatch(n_block_rows,
+                                  block_sparsity_pattern.n_block_rows()));
+      Assert(n_block_rows == block_sparsity_pattern.n_block_cols(),
+             ExcDimensionMismatch(n_block_rows,
+                                  block_sparsity_pattern.n_block_cols()));
+#  endif
+
+
+      // Call the other basic reinit function, ...
+      reinit(block_sparsity_pattern.n_block_rows(),
+             block_sparsity_pattern.n_block_cols());
+
+      // ... set the correct sizes, ...
+      this->row_block_indices    = block_sparsity_pattern.get_row_indices();
+      this->column_block_indices = block_sparsity_pattern.get_column_indices();
+
+      // ... and then assign the correct
+      // data to the blocks.
+      for (size_type r = 0; r < this->n_block_rows(); ++r)
+        for (size_type c = 0; c < this->n_block_cols(); ++c)
+          {
+            this->sub_objects[r][c]->reinit(parallel_partitioning[r],
+                                            parallel_partitioning[c],
+                                            block_sparsity_pattern.block(r, c),
+                                            communicator,
+                                            exchange_data);
+          }
+    }
+
+
+
+    template <typename Number, typename MemorySpace>
+    template <typename BlockSparsityPatternType>
+    void
+    BlockSparseMatrix<Number, MemorySpace>::reinit(
+      const BlockSparsityPatternType &block_sparsity_pattern)
+    {
+      std::vector<IndexSet> parallel_partitioning;
+      for (size_type i = 0; i < block_sparsity_pattern.n_block_rows(); ++i)
+        parallel_partitioning.emplace_back(
+          complete_index_set(block_sparsity_pattern.block(i, 0).n_rows()));
+
+      reinit(parallel_partitioning, block_sparsity_pattern);
+    }
+
+
+
+    // template <typename Number, typename MemorySpace>
+    // template<>
+    // void
+    // BlockSparseMatrix<Number, MemorySpace>::reinit(const BlockSparsityPattern
+    // &block_sparsity_pattern)
+    //{
+    //   // Call the other basic reinit function, ...
+    //   reinit(block_sparsity_pattern.n_block_rows(),
+    //          block_sparsity_pattern.n_block_cols());
+
+    //  // ... set the correct sizes, ...
+    //  this->row_block_indices    = block_sparsity_pattern.get_row_indices();
+    //  this->column_block_indices =
+    //  block_sparsity_pattern.get_column_indices();
+
+    //  // ... and then assign the correct
+    //  // data to the blocks.
+    //  for (size_type r = 0; r < this->n_block_rows(); ++r)
+    //    for (size_type c = 0; c < this->n_block_cols(); ++c)
+    //      {
+    //        this->sub_objects[r][c]->reinit(block_sparsity_pattern.block(r,
+    //        c));
+    //      }
+    //}
```

Comment:
I just skipped over this function since it was not necessary for a working set and then forgot about it. 
I have another look into this function. If there is no straight-up way to implement it, I will delete it.